File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/ByteArrayConverter.java
package org.matilda.commands.types;

import com.google.protobuf.Any;
import com.google.protobuf.ByteString;
import com.google.protobuf.BytesValue;
import com.google.protobuf.InvalidProtocolBufferException;

public class ByteArrayConverter implements ProtobufConverter<byte[]> {
    private final ByteStringConverter mByteStringConverter;

    public ByteArrayConverter() {
        mByteStringConverter = new ByteStringConverter();
    }

    @Override
    public BytesValue convertToProtobuf(byte[] object) {
        return mByteStringConverter.convertToProtobuf(ByteString.copyFrom(object));
    }

    @Override
    public byte[] convertFromProtobuf(Any object) throws InvalidProtocolBufferException {
        return mByteStringConverter.convertFromProtobuf(object).toByteArray();
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/ByteArrayConverter.kt
package org.matilda.commands.types

import com.google.protobuf.Any
import com.google.protobuf.ByteString
import com.google.protobuf.BytesValue

class ByteArrayConverter : ProtobufConverter<ByteArray> {
    private val mByteStringConverter = ByteStringConverter()

    override fun convertToProtobuf(obj: ByteArray): BytesValue {
        return mByteStringConverter.convertToProtobuf(ByteString.copyFrom(obj))
    }

    override fun convertFromProtobuf(obj: Any): ByteArray {
        return mByteStringConverter.convertFromProtobuf(obj).toByteArray()
    }
}
